rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java

```diff
@@ -509,6 +516,51 @@ void logPegoutConfirmed() {
         assertEvent(eventLogs, 0, BridgeEvents.PEGOUT_CONFIRMED.getEvent(), new Object[]{btcTx.getHash().getBytes()}, new Object[]{pegoutCreationRskBlockNumber});
     }
 
+    private static Stream<Arguments> logPegoutTransactionCreatedValidArgProvider() {
+        List<Arguments> args = new ArrayList<>();
+
+        args.add(Arguments.of(Sha256Hash.ZERO_HASH, Collections.singletonList(Coin.SATOSHI)));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(5), coinListOf(500_000, 400_000, 1_000)));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(10), Collections.singletonList(Coin.ZERO)));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(10), Collections.EMPTY_LIST));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(15), null));
+
+        return args.stream();
+    }
+
+    @ParameterizedTest()
+    @MethodSource("logPegoutTransactionCreatedValidArgProvider")
+    void logPegoutTransactionCreated_ok(Sha256Hash btcTxHash, List<Coin> outpointValues) {
+        eventLogger.logPegoutTransactionCreated(btcTxHash, outpointValues);
+        commonAssertLogs(eventLogs);
+
+        assertTopics(2, eventLogs);
+
+        int index = 0;
+        Function expectedEvent = BridgeEvents.PEGOUT_TRANSACTION_CREATED.getEvent();
+        Object[] topics = {btcTxHash.getBytes()};
+        Object[] params = {UtxoUtils.encodeOutpointValues(outpointValues)};
+        assertEvent(eventLogs, index, expectedEvent, topics, params);
+    }
+
+    private static Stream<Arguments> logPegoutTransactionCreatedInvalidArgProvider() {
+        List<Arguments> args = new ArrayList<>();
+
+        args.add(Arguments.of(null, Collections.singletonList(null), IllegalArgumentException.class));
+        args.add(Arguments.of(null, Collections.singletonList(Coin.SATOSHI), IllegalArgumentException.class));
+        args.add(Arguments.of(Sha256Hash.ZERO_HASH, Collections.singletonList(null), InvalidOutpointValueException.class));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(1), coinListOf(-100), InvalidOutpointValueException.class));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(2), coinListOf(100, -200, 300), InvalidOutpointValueException.class));
+
+        return args.stream();
+    }
+
+    @ParameterizedTest()
+    @MethodSource("logPegoutTransactionCreatedInvalidArgProvider")
+    void logPegoutTransactionCreated_invalidBtcTxHashOrOutpointValues_shouldFail(Sha256Hash btcTxHash, List<Coin> outpointValues, Class<? extends  Exception> expectedException) {
+        assertThrows(expectedException, () -> eventLogger.logPegoutTransactionCreated(btcTxHash, outpointValues));
```

Comment:
```suggestion
    void logPegoutTransactionCreated_invalidBtcTxHashOrOutpointValues_shouldFail(Sha256Hash btcTxHash, List<Coin> outpointValues, Exception expectedException) {
        assertThrows(expectedException.getClass(), () -> eventLogger.logPegoutTransactionCreated(btcTxHash, outpointValues));
```
That looks a bit better

Reply:
This change implies creating an instance of the exception for each entry of the argument provider method. It does not make sense to me to create an instance just to get the class/type of it.